src/test/fuzz/util/threadinterrupt.cpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/util/threadinterrupt.h>
+
+FuzzedThreadInterrupt::FuzzedThreadInterrupt(FuzzedDataProvider& fuzzed_data_provider)
+    : m_fuzzed_data_provider{fuzzed_data_provider}
+{
+}
+
+bool FuzzedThreadInterrupt::interrupted() const
+{
+    return m_fuzzed_data_provider.ConsumeBool();
+}
+
+bool FuzzedThreadInterrupt::sleep_for(Clock::duration)
+{
+    return m_fuzzed_data_provider.ConsumeBool();
```

Comment:
Perhaps in a follow up we can make mocktime accurate to the millisecond and then advance it here to simulate an actual sleep.

Reply:
I like simulating some time passage here.

Changing `static std::atomic<std::chrono::seconds> g_mock_time{};` to a finer precision is indeed out of the scope here.

Following a discussion on IRC I checked that indeed the time is frozen already for fuzz tests so calling `SetMockTime()` here is not going to freeze it (since it is already frozen). I added:

```cpp
SetMockTime(ConsumeTime(m_fuzzed_data_provider)); // Time could go backwards.
```

It could end up increasing the time more than the argument to `sleep_for()` but this might happen during normal operation, so is a good exercise.

It could end up with the time going backwards. If this is undesirable then this may be changed to something like:

```cpp
SetMockTime(NodeClock::now() + ConsumeIntegralInRange(0, the_argument_to_sleep_for * 2));
```